src/com/jdngray77/htmldesigner/backend/BackgroundTask.kt

```diff
@@ -1,25 +1,52 @@
 package com.jdngray77.htmldesigner.backend
 
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+
 /**
  * Runs tasks in the background
  */
-object BackgroundTask {
+object BackgroundTask : Subscriber {
+
+    private val threadPool = Executors.newCachedThreadPool()
+
+    init {
+        EventNotifier.subscribe(this, EventType.USER_EXIT)
+    }
+
 
     fun runOnUIThread(runnable: Runnable) {
         TODO()
     }
 
+    
+    /**
+     * Submits a [runnable] task to the for [threadPool] for execution.
+     */
     fun submit(runnable: Runnable) {
```

Comment:
Add a guard to reject new submissions if `shutdown` has been called.

Reply:
I have added a guard, though I dont think it is neccessary. Upon a shutdown call, the thread pool is instructed to reject any new submissions and executes all remaining tasks that had been submitted before the call.

However, I could be wrong, so I have added the guard as a precatuionary measure for the time being.